builtin/logical/pki/issuing/context.go

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package issuing
+
+import "context"
+
+// IssuerRoleContext combines in a single struct an issuer and a role that we should
+// leverage to issue a certificate along with the
+type IssuerRoleContext struct {
+	context.Context
```

Comment:
the other two pairings in here make sense, but i'm a bit confused why context is in here?  It feels like it's going to get complicated with, eg. storageContext also existing, and it's not really used in this refactor as far as I can see?

Reply:
Agreed there is a little overlap with `storageContext`. Along all the refactorings we are trying to move away from storageContext as it pulls in too many things. This was refactoring around the new enterprise only packages and their new context patterns that we want to start adopting in the future within CE.